index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
My concern is that the basic type is fairly incoherent.

> My understanding is that `Maybe a` is only a Semigroup if `a` is a Semigroup. `Number` is not a Semigroup, therefore `Maybe Number` is not a Semigroup.

I'm still not sure-footed enough in this stuff to respond with authority, but my instinct is different.  First of all, you're acting as though `Maybe` has some clear-cut semantics.  Perhaps it does for the strongly-typed language Haskell.  But when you try to convert this to Javascript, you have to make some choices.  And those are not at all obvious.  But they are your choices, which you will have to express in your implementation.  Choosing to make `Maybe` a Semigroup and/or a Monoid is such a choice, and certainly doesn't seem to be a necessary one.

Secondly, although Hardy Jones suggested that there might be a standard Monoid over `Maybe a`, there is by no means only one.  We can define many different Semigroups and/or Monoids over the same set of `Maybe a`.  As well as that standard Monoid and the alternative I discussed in Gitter, I think this would also qualify as a Monoid:

    Nothing.concat(m); //=> Nothing
    m.concat(Nothing); //=> Nothing
    Just(a).concat(Just(b)); //=> Just(flatten([a, b]))

for some appropriate shallow `flatten` function.

I may be missing something here, but if this is in fact lawful, it also demonstrates that we can create a Semigroup of `Maybe a` even though `a` is not one itself.

So to me, all this begs the question: what is the reason for trying to make `Maybe` (or `Maybe a`) a Semigroup?  What's the usefulness?  I understand the rationale for it being a Functor and a Monad, and hence all the related ADTs as well.  Why do you want it to be a Semigroup/Monoid?